sim/core/spell.go

```diff
@@ -146,6 +146,8 @@ type Spell struct {
 	initialCritMultiplier           float64
 	initialThreatMultiplier         float64
 	// Note that bonus expertise and armor pen are static, so we don't bother resetting them.
+
+	resultCache SpellEffect
```

Comment:
This won't work in the general case, because there are a small number of spells for which there are multiple 'calcs' active at the same time. The examples that come to mind are 1) whenever travel time is long enough and 2) when calculating all aoe spell results before dealing them.

Reply:
I've added an "inUse" bool for SpellEffect that's set in calcDamageInternal(), and reset in dealDamageInternal(). If the resultCache is in use (which only seems to happen for warlocks, with long travel times), a new SpellEffect is created.
I ignored warlock test changes before, because I _always_ get test changes for them on my current setup :/